src/url-params.ts

```diff
@@ -63,12 +63,12 @@ export class AnalyticsUrlParams {
 				passedDeviceId,
 				originalDeviceId,
 			);
-			if (this.client != null && newCurrentDeviceId != null) {
-				this.client.setDeviceId(newCurrentDeviceId);
-			}
+			//if (this.client != null && newCurrentDeviceId != null) {
+			//	this.client.setDeviceId(newCurrentDeviceId);
+			//}
 
 			params.delete(URL_PARAM_DEVICE_ID);
-			return params.toString();
+			return [passedDeviceId, params.toString()]; //Im passing `passedDeviceId` cause I dont know how to set the typescript return type in order to accept a list as returned by setDeviceIds()
```

Comment:
> accept a list as returned by `setDeviceIds()`

`setDeviceIds` returns the first element of the array or `null`, its signature is
```typescript
private AnalyticsUrlParams.setDeviceIds(
    inputIdString: string | null | undefined,
    currentDeviceId: string | null): string | null
```
so an array/list cannot be obtained from it.

Reply:
Right! Thanks Pranas, I missed that.